lib/ui/compositing.dart

```diff
@@ -200,7 +371,13 @@ class SceneBuilder extends NativeFieldWrapperClass2 {
   /// Therefore, when implementing a subclass of the [Layer] concept defined in
   /// the rendering layer of Flutter's framework, once this is called, there's
   /// no need to call [addToScene] for its children layers.
-  void addRetained(EngineLayer retainedLayer) native 'SceneBuilder_addRetained';
+  void addRetained(EngineLayer retainedLayer) {
+    assert(retainedLayer is _EngineLayerWrapper);
+    assert(_debugCheckUsedOnce(retainedLayer, 'addRetained'));
```

Comment:
this also needs to add all the descendants of the retained layer

Reply:
Ah, you're right. Done.